src/tests/Autosuggest.js

```diff
@@ -412,6 +426,106 @@ describe('Autosuggest', function() {
     });
   });
 
+  describe('onSuggestionFocused', function() {
+    beforeEach(function() {
+      onSuggestionFocused.mockClear();
+      onSuggestionUnfocused.mockClear();
+      createAutosuggest(
+        <Autosuggest suggestions={getSuburbStrings}
+                     onSuggestionFocused={onSuggestionFocused}
+                     onSuggestionUnfocused={onSuggestionUnfocused} />
+      );
+      setInputValue('m');
+    });
+
+    describe('Mouse interactions', function() {
+
+      it('should call onSuggestionFocused when suggestion focused using mouse', function() {
+        mouseOverFromInputToSuggestion(1);
+        expect(onSuggestionFocused).toBeCalledWith('Mordialloc');
+        expect(onSuggestionUnfocused).not.toBeCalled();
+      });
+    });
+
+    describe('Keyboard interactions', function() {
+
+      it('should call onSuggestionFocused when suggestion focused using Up/Down keys', function() {
+        clickDown();
+        expect(onSuggestionFocused).toBeCalledWith('Mill Park');
+      });
+
+      it('should not call onSuggestionFocused when first suggestion is focused and Up key is clicked', function() {
+        clickDown();
+        onSuggestionFocused.mockClear();
+        clickUp();
+        expect(onSuggestionFocused).not.toBeCalled();
+      });
+
+      it('should not call onSuggestionFocused when last suggestion is focused and Down key is clicked', function() {
+        clickDown();
+        clickDown();
+        onSuggestionFocused.mockClear();
+        clickDown();
+        expect(onSuggestionFocused).not.toBeCalled();
+      });
+    });
+  });
+
+  describe('onSuggestionUnfocused', function() {
+    beforeEach(function() {
+      onSuggestionFocused.mockClear();
+      onSuggestionUnfocused.mockClear();
+      createAutosuggest(
+        <Autosuggest suggestions={getSuburbObjects}
+                     suggestionRenderer={renderSuburbObject}
+                     suggestionValue={getSuburbObjectValue}
+                     onSuggestionFocused={onSuggestionFocused}
+                     onSuggestionUnfocused={onSuggestionUnfocused} />
+      );
+      setInputValue('m');
+    });
+
+    describe('Mouse interactions', function() {
+
+      it('should call onSuggestionUnfocused when suggestion unfocused using mouse', function() {
+        mouseOverFromInputToSuggestion(0);
+        mouseOverBetweenSuggestions(0, 1);
+        expect(onSuggestionUnfocused).toBeCalledWith({ suburb: 'Mill Park', postcode: '3083' });
+      });
+
+      it('should call onSuggestionUnfocused when clicking outside and suggestion is focused', function() {
+        mouseOverFromInputToSuggestion(0);
+        clickOutside();
+        expect(onSuggestionUnfocused).toBeCalledWith({ suburb: 'Mill Park', postcode: '3083' });
+      });
+
+      it('should not call onSuggestionUnfocused when clicking outside and no suggestion focused', function() {
+        clickOutside();
+        expect(onSuggestionUnfocused).not.toBeCalled();
+      });
```

Comment:
I suggest we move the above 2 tests to the `'Keyboard interactions'` section below because you can't really click outside without moving the mouse from the suggestion to the outside area first. It makes more sense to me to check a scenario when you:
1. click Down to focus on suggestion
2. click outside
3. verify onSuggestionUnfocused is called
or 
1. click Down
2. click Down
3. click Down (now focus should go back to the input field and no suggestion is focused)
4. click outside
5. verify onSuggestionUnfocused is not called